ibis/common/graph.py

```diff
@@ -88,19 +84,19 @@ def _recursive_get(obj: Any, dct: dict, filter: type) -> Any:
     >>> from ibis.common.graph import _recursive_get
     >>>
     >>> dct = {1: 2, 3: 4}
-    >>> _recursive_get((1, 3), dct, filter=int)
+    >>> _recursive_get((1, 3), dct)
     (2, 4)
-    >>> _recursive_get(frozendict({1: 3}), dct, filter=int)
+    >>> _recursive_get(frozendict({1: 3}), dct)
     {1: 4}
-    >>> _recursive_get(frozendict({1: (1, 3)}), dct, filter=int)
+    >>> _recursive_get(frozendict({1: (1, 3)}), dct)
     {1: (2, 4)}
     """
-    if isinstance(obj, filter):
-        return dct[obj]
+    if isinstance(obj, Node):
+        return dct.get(obj, obj)
```

Comment:
Not knowing much else about the intent here, I'd say this `get` should return `None` since this function is called `_recursive_get`, typically returning the `Node` doesn't match the input mapping's value type, making returning a `Node` very surprising when the lookup fails IMO.

Reply:
Well, we need to keep the functionality as is, so it would better to rename the function instead. 

After this PR, when we stop the traversal at a `node`, its children are not going to be part of the graph, hence they won't be added to the results dictionary, but we still need to be able to recreate the `node` where we need the original children.

Reply:
Naming suggestion? e.g. `_recursive_lookup`?

Reply:
Renamed.